tests/core/encoding.c

```diff
@@ -0,0 +1,33 @@
+#include "clar_libgit2.h"
+#include "varint.h"
+
+void test_core_encoding__decode(void)
+{
+	const unsigned char *buf = (unsigned char *)"AB";
+	size_t size;
+
+	cl_assert(git_decode_varint(buf, &size) == 65);
+	cl_assert(size == 1);
+
+	buf = (unsigned char *)"\xfe\xdc\xbaXY";
+	cl_assert(git_decode_varint(buf, &size) == 267869656);
+	cl_assert(size == 4);
+
+	buf = (unsigned char *)"\xaa\xaa\xfe\xdc\xbaXY";
+	cl_assert(git_decode_varint(buf, &size) == 1489279344088ULL);
+	cl_assert(size == 6);
+
```

Comment:
For completeness, I'd also add a test where parsing actually fails so that `size == 0`.